src/main/java/parsing/Parser.java

```diff
@@ -97,14 +97,49 @@ public NodeGraph parse(final File file, NodeGraph graph) {
             BufferedWriter out = new BufferedWriter(new FileWriter(segments));
             BufferedWriter gw = new BufferedWriter(new FileWriter(genomes));
 
-            addGenomes(gw, line);
```

Comment:
I see there was a method here. You replaced it with a lot of extra lines in this method which makes this one a lot longer. Perhaps you could move the added code to a seperate method.